### ISSUE TITLE
Add warning (or find a way) of doing cursor checks instead of deep-equal checks

### ISSUE THREAD

**Issue by mikaelbr** [omniscientjs/omniscient#39]
Sometimes (e.g. when namespacing), cursors are checked with a isEqual check. This needs to be working better. At least show a warning and clear the usage. Preferably, it should just work, but it depends on performance and complexity.

1. Verify problem by adding tests
2. Fix issue by warning or having a isCursor test before equal test in some way.

**Comment by Dashed** [comment]
@mikaelbr May need to provide `customizer` function to `_.isEqual`: https://lodash.com/docs#isEqual

`customizer` would return `undefined` for anything that's not a cursor; bool otherwise.

**Comment by torgeir** [comment]
By namespacing you mean cursors nested several levels deep?

**Comment by mikaelbr** [comment]
Correct. Having `Component({ cursors: { cursor1, cursor2 } })`

**Comment by Dashed** [comment]
Thinking on this some more, I think there can be more done with `shouldComponentUpdate` so that the comparison process is done uniformly on both `props` and `state`. At the moment, `state` just gets the `isEqual` treatment.

Here's an example of putting a cursor in component's state:

```js
var structure = immstruct({turtles: []});

var TurtlesComponent = React.createClass({

  componentWillMount: function() {
    this.setState({turtlesCursor: structure.cursor()}

    structure.on('swap', () =>
      this.setState({turtlesCursor: structure.cursor()}));

  },

  onAddButtonClick: function(e) {
    this.state.turtlesCursor.updateIn('turtles', (turtles) => 
      turtles.push(e.target.turtle)
    );
  },

  renderTurtleCount: function() {
    return <div>{this.state.turtlesCursor.deref().size}</div>;
    // or pass own sub-cursor to a child component...
  }

});
```

**Comment by mikaelbr** [comment]
> @mikaelbr May need to provide customizer function to _.isEqual: https://lodash.com/docs#isEqual
> customizer would return undefined for anything that's not a cursor; bool otherwise.

That is actually a brilliant idéa, @Dashed ! I'm trying it out now. Seems to work perfectly.

**Comment by Dashed** [comment]
Awesome! Looking forward to see how it works out.

**Comment by mikaelbr** [comment]
Fixed in commit 46c25ced317056ea14f90a555a44d5e2000c55df

**Comment by Dashed** [comment]
Nice. Looking better now. I haven't yet look through the tests fully yet. Might do it over the weekend.
